src/tests/utils/provider-util.tsx

```diff
@@ -1,14 +1,15 @@
 import React, { ReactElement } from 'react'
 import { render, RenderOptions } from '@testing-library/react'
-import { Provider } from '../rollbar-react'
+import { Provider, ProviderProps } from '../rollbar-react'
 
 export const renderWithProviderProps = (
   ui: ReactElement,
   options: RenderOptions,
-  providerProps: any
-) => {
-  return render(ui, {
-    wrapper: (props) => <Provider {...props} {...providerProps} />,
+  providerProps: Omit<ProviderProps, 'children'>
+) =>
+  render(ui, {
+    wrapper: (props) => (
+      <Provider {...{ children: props.children ?? <></> }} {...providerProps} />
+    ),
```

Comment:
This one was a fun type-structure puzzle.

So, `Provider` expects a `ProviderProps`-type strcture which is defined like so:
```
export interface ProviderProps {
  Rollbar?: new (options: Configuration) => Rollbar
  children: ReactNode
  config?: Configuration | (() => Configuration)
  instance?: Rollbar
}
```

The solution is two-fold.

First we need to acknowledge that we're providing the expected type structure through two arguments, `props` supplies the only ***required*** prop `children`, and `providerProps` supplies the rest of the props which are all optional.

Second problem is a bit harder to explain. The function render is defined as: `render(_, RenderOptions)`, we only care about the second argument. `RenderOptions` has a bunch of props and `wrapper?: React.ComponentType` which is the one we care about.

ComponentType is defined as:
```ts
type ComponentType<P = {}> = ComponentClass<P> | FunctionComponent<P>;
```
Since we're passing a function `(props) => ...`, we are matching `FunctionComponent`'s structure, which has a bunch of props and the main one, the function, which is defined as:
```ts
interface FunctionComponent<P = {}> {
  (props: PropsWithChildren<P>, context?: any): ReactElement<any, any> | null;
  ...
}  
```

So, the props we get here: `(props) => <Provider {...props} {...providerProps} />,` is of type `PropsWithChildren` which is defined:
```ts
type PropsWithChildren<P> = P & { children?: ReactNode | undefined };
```

And therein lies the second problem. The `children` prop is optional, but `ProviderProps` requires it. So, we solve this by creating a new object that matches the structure in `ProviderProps`, and we either pass `props.children` if there's something in it, or an empty node.

Fun for the whole family.